web/src/stream_data.ts

```diff
@@ -744,29 +798,20 @@ export function create_sub_from_server_data(attrs) {
     delete attrs.subscribers;
 
     sub = {
-        name: attrs.name,
-        render_subscribers: !page_params.realm_is_zephyr_mirror_realm || attrs.invite_only === true,
-        subscribed: true,
+        render_subscribers: !page_params.realm_is_zephyr_mirror_realm || attrs.invite_only,
         newly_subscribed: false,
         is_muted: false,
-        invite_only: false,
         desktop_notifications: null,
         audible_notifications: null,
         push_notifications: null,
         email_notifications: null,
         wildcard_mentions_notify: null,
-        description: "",
-        rendered_description: "",
-        first_message_id: attrs.first_message_id,
+        color: "color" in attrs ? attrs.color : color_data.pick_color(),
```

Comment:
`pick_color` has a side effect of modifying `color_data.unused_colors`, so we shouldn’t speculatively call it here to build the defaults.

Reply:
It should only be called when we indeed don't have a "color" property present in `attrs` object otherwise it will use the given value.

Also in previous version of code we were setting `sub.color` and calling `pick_color` just after creating this `sub` object, like this and I think the current ternary I wrote basically does the same thing right?  
https://github.com/zulip/zulip/blob/6632eca2dcf2051043e95348658d6da02305669f/web/src/stream_data.js#L762-L764